src/libraries/System.Security.Cryptography.Algorithms/tests/ChaCha20Poly1305Tests.cs

```diff
@@ -447,6 +448,11 @@ public static void CheckIsSupported()
                 // The test queries the OS directly to ensure our version check is correct.
                 expectedIsSupported = CngUtility.IsAlgorithmSupported("CHACHA20_POLY1305");
             }
+            else if (PlatformDetection.IsAndroid)
+            {
+                // Android with API Level 28 is the minimum API Level support for ChaChaPoly1305.
+                expectedIsSupported = GetAndroidSdkVersion() >= 28;
```

Comment:
I'd rather we import the Android interop files to do the `Interop.Crypto.CipherIsSupported(Interop.Crypto.EvpChaCha20Poly1305());` check instead of the Android API version. Importing the interop APIs is the same pattern used with AES, RC2, etc.

Reply:
Hm. Maybe that kind of defeats the purpose of this test.

The tests are already conditional on ChaChaPoly1305.IsSupported which relies on `CipherIsSupported`. The purpose of this test is to test the `IsSupported` property against some other measure of "is this cipher supported"?

Reply:
That said, I am happy to make the change if we are sure testing the `IsSupported` property against the API Level is not desirable.

Reply:
I missed that this was a test of the IsSupported property itself. This is a good way of doing it then I think.